tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/PipelineConvention.cs

```diff
@@ -249,7 +249,7 @@ private bool EnsureReportBuildStatus(BuildDefinition definition)
         /// <summary>
         /// Number of buckets for hour hashing
         /// </summary>
-        protected const int HourBuckets = 3;
+        protected const int HourBuckets = 12;
```

Comment:
Can we limit this to 8 hours and update the StartHourOffset to 0?

Reply:
That works for me ... how about you @danieljurek?

Reply:
As we chatted we should keep the 12 hour window and start the offset at 0 so things stay on the same day. We should change the bucketing to be based on the build definition id and be spaced every 15 mins.

Reply:
OK so the way I approached this to get a distribution across 12 hours (roughly) was as follows:

1. Took the start and finish time hours and calculated the total minutes.
2. Sliced that up into 15 minute buckets to calculate the # of buckets.
3. Did a modulo operation on the definition ID to slot it into a bucket to give some kind of distribution, not even but probably good enough.
4. Converted the bucket back into an hour/minute combo for the schedule.

I also lifted the schedule trigger logic into the base class and created an overridable property so that we can turn off schedules for PR pipelines rather than duplicating the scheduled trigger creation.

I also refined the conditions in which the scheduled trigger is recreated so that if we change the algo for bucketing that the schedules will get updated.